e2e/step_definitions/programCatalog.js

```diff
@@ -17,6 +17,22 @@ defineSupportCode(({
         return ProgramCatalog.load();
     });
 
+    When(/^the course count is remembered$/, () => {
+        return ProgramCatalog.countCourses().then(count => {
+            ProgramCatalog.courseCount = count;
+        });
+    });
+
+    When(/^the search bar is cleared$/, () => {
+        ProgramCatalog.clearSearchBar();
+        return browser.sleep(5000);
+    });
+
+    When(/^the "([^"].*)" dropdown is clicked$/, text => {
+        ProgramCatalog.openFilterDropdown(text);
+        return browser.sleep(2000);
```

Comment:
Please use a wait fuction here, instead of a sleep, which waits until the drop down list becomes visible.